src/utils/sshUtils.ts

```diff
@@ -14,21 +14,28 @@ import { cpUtils } from "./cpUtils";
 
 export const sshFsPath: string = join(os.homedir(), '.ssh');
 
-export async function getSshKey(vmName: string, passphrase: string): Promise<string> {
+export async function createSshKey(vmName: string, passphrase: string): Promise<{ sshKeyName: string; keyData: string }> {
     return await callWithMaskHandling(
         async () => {
-            const sshKeyName: string = `azure_${vmName}_rsa`;
-            const sshKeyPath: string = join(sshFsPath, sshKeyName);
+            let sshKeyName: string = `azure_${vmName}_rsa`;
+            let sshKeyPath: string = join(sshFsPath, sshKeyName);
+            let keyExists: boolean = await fse.pathExists(`${sshKeyPath}.pub`);
 
-            if (!await fse.pathExists(`${sshKeyPath}.pub`)) {
-                ext.outputChannel.appendLog(localize('generatingKey', 'Generating public/private rsa key pair in "{0}"...', sshKeyPath));
-                // create the .ssh folder if it doesn't exist
-                await fse.ensureDir(sshFsPath);
-                await cpUtils.executeCommand(undefined, undefined, 'ssh-keygen', '-t', 'rsa', '-b', '4096', '-f', cpUtils.wrapArgInQuotes(sshKeyPath), '-N', cpUtils.wrapArgInQuotes(passphrase));
-                ext.outputChannel.appendLog(localize('generatedKey', 'Generated public/private rsa key pair in "{0}".', sshKeyPath));
+            while (keyExists) {
+                let count: number = 2;
+                sshKeyName = `azure_${vmName}_${count}_rsa`;
+                sshKeyPath = join(sshFsPath, sshKeyName);
+                keyExists = await fse.pathExists(`${sshKeyPath}.pub`);
```

Comment:
I think you should always have a timeout or [maxTries](https://github.com/microsoft/vscode-azuretools/blob/main/ui/src/wizard/AzureNameStep.ts#L29) for a while loop. It's a little more tedious, but I think it's worth it to reduce the risk of a while loop going on forever (which is one of the worst bugs we can cause IMO). Unfortunately that relies on the "count" variable being accurate, so it doesn't help the bug mentioned above, but still 😝